### PR TITLE
refactor: deprecate button reason

### PR DESCRIPTION
**Pull request by veegandhi**
## Description

Internally in `Bootstrapbutton`, we wrap `Popover` so user can show a message when button is displayed. This introduces an extra layer of wrapping for the component. Popover are display `inline-flex`, this also makes button styling break intermittently based on parent container's display properties. And putting icons within buttons require more customized styles for each button. This makes is difficult for general use case (which is priority). 

Originally it was decided to conditionally wrap this with popover, however, we now provide an extensive api of Popover so it would be easy for user to implement this.

Hence deprecating `reason` prop from `Button`.

Fixes: #849

## Does this PR introduce a breaking change?

- [ ] Yes
- [X] No

### CODE REVIEW

File: src/components/third-party/Button/index.jsx [hunk 1]
```diff
@@ -21,14 +19,6 @@ class Button extends React.PureComponent {
   constructor(props) {
     super(props);
     this.buttonRef = React.createRef();
```
Comment (lightbringer1991) [inline src/components/third-party/Button/index.jsx]: do we still need this ref?

File: www/examples/ButtonExample.jsx [hunk 2]
```diff
@@ -13,14 +13,20 @@ class ButtonExample extends React.PureComponent {
...
+        >
+          <Button bsStyle="link" disabled={!this.state.canUndo} onClick={this.onClick}>
+            Undo
+          </Button>
+        </Popover>
         <Button bsStyle="primary" disabled={this.state.canUndo} onClick={this.onClick}>
           Apply
         </Button>
-      </div>
+      </React.Fragment>
```
Comment (lightbringer1991) [inline www/examples/ButtonExample.jsx]: I think we can even remove this example. There's no need to show usage of Popover with button

File: src/components/third-party/Button/index.jsx (hunk 1 above)
Comment (omgaz) [reply src/components/third-party/Button/index.jsx]: If not, do we still need the constructor function: that can probably go as well (also hi)

File: www/mdexamples/TreePicker.mdx [hunk 3]
```diff
@@ -83,7 +83,7 @@ const Example = () => {
         onChange={setPickerSearchValue}
         searchOnClear={() => setPickerSearchValue('')}
         includeNode={node => setSelectedNodes(_.concat([], selectedNodes, node))}
-        removeNode={node => setSelectedNodes(_.reject(this.state.selectedNodes, { id: node.id }))}
+        removeNode={node => setSelectedNodes(_.reject(selectedNodes, { id: node.id }))}
```
Comment (Harris-YWang) [inline www/mdexamples/TreePicker.mdx]: fix a tiny bug on the ui.adslot.com page

File: src/components/Button/index.jsx [hunk 4]
```diff
@@ -7,79 +7,40 @@ import PropTypes from 'prop-types';
...
-          {this.renderSpinner()}
-          <div className={isLoading ? 'button-component-children-container' : null}>{children}</div>
-        </Popover>
-      </BootstrapButton>
-    );
-  }
-}
+const Button = props => {
+  const { inverse, children, dts, className, isLoading, disabled } = props;
+  const baseClass = 'button-component';
```
Comment (superpower1) [inline src/components/Button/index.jsx]: I think we are going to give all ui components a `aui--` prefix, so maybe change it to `aui--button`?

Comment (lightbringer1991) [reply src/components/Button/index.jsx]: this change will lead to some style breaking in other libraries as well. But I'd say let's do it.

Comment (Harris-YWang) [reply src/components/Button/index.jsx]: Fixed and checked relevant files in the project. Any further idea about other libraries? Further changes would be made if we need.

File: src/components/Button/index.jsx [hunk 5]
```diff
@@ -7,79 +7,40 @@ import PropTypes from 'prop-types';
...
-        </Popover>
-      </BootstrapButton>
-    );
-  }
-}
+const Button = props => {
+  const { inverse, children, dts, className, isLoading, disabled } = props;
+  const baseClass = 'aui--button';
+  const classes = classNames(baseClass, className, {
+    'btn-inverse': inverse && !/btn-inverse/.test(className),
```
Comment (lightbringer1991) [inline src/components/Button/index.jsx]: @veegandhi just wondering if we want to go all BEM, e.g. `aui--button__inverse`, `aui--button-spinner`, etc.

Comment (veegandhi) [reply src/components/Button/index.jsx]: Not really. For not lets just hyphenate the class names.

**Pull request merged**